yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+            'upload_date': '20151118',
+            'creator': 'David Tanis',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1024781-neapolitan-checkerboard-cookies',
+        'md5': '58df35998241dcf0620e99e646331b42',
+        'info_dict': {
+            'id': '1024781',
+            'ext': 'mp4',
+            'title': 'Neapolitan Checkerboard Cookies',
+            'description': 'md5:ba12394c585ababea951cb6d2fcc6631',
+            'timestamp': 1701302400,
+            'upload_date': '20231130',
+            'creator': 'Sue Li',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1019516-overnight-oats',
+        'md5': '2fe7965a3adc899913b8e25ada360823',
+        'info_dict': {
+            'id': '1019516',
+            'ext': 'mp4',
+            'timestamp': 1546387200,
+            'description': 'md5:8856ce10239161bd2596ac335b9f9bfb',
+            'upload_date': '20190102',
+            'title': 'Overnight Oats',
+            'creator': 'Genevieve Ko',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        }
+    }]
+
     def _real_extract(self, url):
         page_id = self._match_id(url)
-
         webpage = self._download_webpage(url, page_id)
 
-        video_id = self._search_regex(
-            r'data-videoid=["\'](\d+)', webpage, 'video id',
-            default=None, fatal=False)
-        if video_id is not None:
-            return self._extract_video_from_id(video_id)
+        json_obj = traverse_obj(
+            self._search_nextjs_data(webpage, page_id), ('props', 'pageProps'), default={})
 
-        podcast_data = self._search_regex(
-            (r'NYTD\.FlexTypes\.push\s*\(\s*({.+?})\s*\)\s*;\s*</script',
-             r'NYTD\.FlexTypes\.push\s*\(\s*({.+})\s*\)\s*;'),
-            webpage, 'podcast data')
-        return self._extract_podcast_from_json(podcast_data, page_id, webpage)
+        info = traverse_obj(json_obj, {
+            'id': ('recipe', 'id', {str_or_none}),
+            'title': ('recipe', 'title'),
+            'description': ('recipe', 'topnote', {clean_html}),
+            'timestamp': ('recipe', 'publishedAt'),
+            'creator': ('recipe', 'contentAttribution', 'cardByline'),
+            'upload_date': ('meta', 'jsonLD', 'video', 'uploadDate', {unified_strdate}),
+            'formats': ('recipe', 'videoSrc', {url_or_none}),
+        })
 
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            info.get('formats'), info.get('id'), 'mp4', entry_protocol='m3u8_native', m3u8_id='hls')
+
+        thumbnails = []
+        for image in traverse_obj(json_obj, ('recipe', 'image', 'crops', 'recipe', ...)):
+            if not url_or_none(image):
+                continue
+            thumbnails.append({
+                'url': image,
+            })
```

Comment:
this can be inlined in the return too, e.g.
```py
            'thumbnails': traverse_obj(next_data, (
                'recipe', 'image', 'crops', 'recipe', lambda _, v: url_or_none(v['url']), {
                    'url': 'url',
                })),
```

Reply:
The thumbnails returned in the block don't have `url` as key. This is a sample output:
```
'XL':
'https://static01.nyt.com/images/2015/11/13/dining/13TANISDESERT1/13DESERT1-master768.jpg'
'L':
'https://static01.nyt.com/images/2015/11/13/dining/13TANISDESERT1/13DESERT1-articleLarge.jpg'
'M':
'https://static01.nyt.com/images/2015/11/13/dining/13TANISDESERT1/13DESERT1-master768.jpg'
'S':
'https://static01.nyt.com/images/2015/11/13/dining/13TANISDESERT1/13DESERT1-blog480.jpg'
'XS':
'https://static01.nyt.com/images/2015/11/13/dining/13TANISDESERT1/13DESERT1-tmagSF.jpg'
```
I modified your suggestion to this:
```python
'thumbnails': [{'url': url} for url in traverse_obj(next_data, (
    'recipe', 'image', 'crops', 'recipe', lambda _, v: url_or_none(v)))]
```
This also works:
```python
'thumbnails': [{'url': url} for url in traverse_obj(next_data, (
              'recipe', 'image', 'crops', 'recipe', ..., {url_or_none}))
```

I assume there is a better way to do this.

Reply:
I'd use the second, that looks fine